presto-main/src/main/java/io/prestosql/operator/scalar/StringFunctions.java

```diff
@@ -195,14 +196,70 @@ public static Slice reverse(@SqlType("varchar(x)") Slice slice)
     @SqlType(StandardTypes.BIGINT)
     public static long stringPosition(@SqlType("varchar(x)") Slice string, @SqlType("varchar(y)") Slice substring)
     {
+        return stringPosition(string, substring, 1);
+    }
+
+    @Description("returns index of n-th occurrence of a substring (or 0 if not found)")
+    @ScalarFunction("strpos")
+    @LiteralParameters({"x", "y"})
+    @SqlType(StandardTypes.BIGINT)
+    public static long stringPosition(@SqlType("varchar(x)") Slice string, @SqlType("varchar(y)") Slice substring, @SqlType(StandardTypes.BIGINT) long instance)
+    {
+        if (instance < 0) {
+            return stringPositionFromEnd(string, substring, abs(instance));
+        }
+        return stringPositionFromStart(string, substring, instance);
+    }
+
+    private static long stringPositionFromStart(Slice string, Slice substring, long instance)
+    {
+        if (instance <= 0) {
+            throw new PrestoException(INVALID_FUNCTION_ARGUMENT, "'instance' must be a positive or negative number.");
+        }
         if (substring.length() == 0) {
             return 1;
         }
 
-        int index = string.indexOf(substring);
-        if (index < 0) {
-            return 0;
+        int foundInstances = 0;
+        // set the initial index just before the start of the string
```

Comment:
These comments are not very useful, as they just describe what the code is doing. As a general rule, document what's not obvious from the immediate code (hidden relationships, assumptions, non-obvious behaviors, etc).